data/src/message.rs

```diff
@@ -155,7 +155,8 @@ impl Target {
 #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 pub enum Direction {
     Sent,
-    Received,
+    /// is_echo
+    Received(bool),
```

Comment:
@andymandias This breaks `history`. All my messages got deleted :(

We need to make sure we add things in a backwards compatible way


This means we need a custom De-serialize implementation for `Direction`